### PR TITLE
[fix]: [OIP-565]: task duration formatting & added task first picked at for data collection task , state machine

### PR DESCRIPTION
**Pull request by n00bitax**
## Describe your changes
 Added task first picked At for data collection tasks and analysis state machine
 Changed date to ISO format.
 Added test cases.
 
Screenshot 2023-07-11 at 8 16 59 AM

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50306)

### CODE REVIEW

**Review by dhruv-harness — DISMISSED** [review]

**Comment by n00bitax** [comment]
trigger uts

**Comment by n00bitax** [comment]
trigger uts

**Comment by n00bitax** [comment]
trigger ut6

**Comment by n00bitax** [comment]
trigger ut9

**Comment by n00bitax** [comment]
trigger uts

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/DataCollectionTaskServiceImpl.java [hunk 1]
```diff
@@ -95,13 +97,30 @@ public Optional<DataCollectionTask> getNextTask(String accountId, String dataCol
...
   }
 
+  private void updateFirstPickedAt(
+      String accountId, String dataCollectionWorkerId, Instant instant, DataCollectionTask task) {
+    if (Objects.nonNull(task) && task.getFirstPickedAt() == null) {
+      Query<DataCollectionTask> updateTaskUUIDFilter =
+          hPersistence.createQuery(DataCollectionTask.class)
+              .filter(DataCollectionTaskKeys.accountId, accountId)
+              .filter(DataCollectionTaskKeys.dataCollectionWorkerId, dataCollectionWorkerId)
+              .filter(DataCollectionTaskKeys.status, DataCollectionExecutionStatus.RUNNING)
```
Comment (naveen-harness) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/DataCollectionTaskServiceImpl.java:115]: I think we don't need other filters if we're querying by UUID ?

Comment (n00bitax) [reply srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/DataCollectionTaskServiceImpl.java:115]: The RUNNING do need to be kept as a guard rail, I think we can removed the other two. Will take it later.

Comment (naveen-harness) [reply srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/DataCollectionTaskServiceImpl.java:115]: 👍

Comment (n00bitax) [reply srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/DataCollectionTaskServiceImpl.java:115]: nvm uuid doesn't have index, we cant till we create an index for uuid.

**Review by dhruv-harness — APPROVED** [review]

**Pull request merged**